rqt_py_common/src/rqt_py_common/topic_helpers.py

```diff
@@ -244,10 +244,29 @@ def _get_field_type(topic_names_and_types, target):  # noqa: C901
                     except ValueError:
                         pass
 
-    logger.debug('faild to find field type: {}'.format(target))
+    logger.debug('failed to find field type: {}'.format(target))
     return None, False
 
 
+def get_property_type(message_class, property_path):
+    """
+    Get the Python type of a specific property in the given message class.
+
+    If the property is an array, the type of the array's values are returned and the
+    is_array flag is set to True. This is a static type check, so it works for
+    unpublished topics and with empty arrays.
+
+    :param message_class: message class type, ``type``
+    :param property_path: path to the property inside the message class, ``str``,
+        i.e. 'header/seq'
+    :returns: property_type, is_array
+    """
+    is_array = False
+    properties = [f for f in property_path.split('/') if f]
+    slot_path = '/'.join(['_' + p for p in properties])
```

Comment:
are we still using underscores? the new changes to rosidl_python should make this cleaner https://github.com/ros2/rosidl_python/pull/19

Reply:
I'll update based on that new PR